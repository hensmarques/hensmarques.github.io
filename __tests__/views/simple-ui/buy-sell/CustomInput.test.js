import React from 'react';
import { CustomInput } from '../../../../src/views/simple-ui/buy-sell/CustomInput';

describe('<CustomInput/>', () => {
  const mockProps = {
    selectedInstrument: {
      base: 'BTC',
      quote: 'USD'
    },
    basevalue: 1,
    quotevalue: 3000,
    transactionType: 'buy'
  };
  const translationSpy = jest.spyOn(window, '_t');

  it('renders the custom input component for a buy transaction and matches the snapshot', () => {
    const wrapper = shallow(<CustomInput {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the custom input component for a sell transaction and matches the snapshot', () => {
    mockProps.transactionType = 'sell';
    const wrapper = shallow(<CustomInput {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles translations', () => {
    expect(translationSpy).toHaveBeenCalledWith('Select Amount to {action}', 'BUY_SELL_CUSTOM.SELECT_TITLE', {action: ' buy'});
    expect(translationSpy).toHaveBeenCalledWith('buy', 'BUY_SELL_CUSTOM.BUY');
    expect(translationSpy).toHaveBeenCalledWith(
      'Enter the amount of {base} you wish to {action}. You may enter the amount either directly in {base} units, or in the equivalent amount of {quote}.',
      'BUY_SELL_CUSTOM.AMOUNT_TEXT_TWO',
      { base: 'BTC', quote: 'USD', action: ' buy' }
    );
    expect(translationSpy).toHaveBeenCalledWith('Select Amount to {action}', 'BUY_SELL_CUSTOM.SELECT_TITLE', {action: ' sell'});
    expect(translationSpy).toHaveBeenCalledWith('sell', 'BUY_SELL_CUSTOM.SELL');
    expect(translationSpy).toHaveBeenCalledWith(
      'Enter the amount of {base} you wish to {action}. You may enter the amount either directly in {base} units, or in the equivalent amount of {quote}.',
      'BUY_SELL_CUSTOM.AMOUNT_TEXT_TWO',
      { base: 'BTC', quote: 'USD', action: ' sell' }
    );
  });
});
