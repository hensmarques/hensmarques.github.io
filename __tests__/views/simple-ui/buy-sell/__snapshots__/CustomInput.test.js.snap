// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomInput/> renders the custom input component for a buy transaction and matches the snapshot 1`] = `
<div>
  <h3>
    Select Amount to  buy
  </h3>
  <label>
    BTC
  </label>
  <TextInput
    base="base"
    className="input-quantity-custom "
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="basevalue"
    onChange={[Function]}
    placeholder="0"
    placeholderParam={Object {}}
    translation="BUY_SELL_CUSTOM"
    type="text"
    value="1"
    writeOnce={false}
  />
  <label>
    USD
  </label>
  <TextInput
    className=""
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="quotevalue"
    onChange={[Function]}
    placeholder="0"
    placeholderParam={Object {}}
    translation="BUY_SELL_CUSTOM"
    type="text"
    value="3000"
    writeOnce={false}
  />
  <p>
    Enter the amount of BTC you wish to  buy. You may enter the amount either directly in BTC units, or in the equivalent amount of USD.
  </p>
</div>
`;

exports[`<CustomInput/> renders the custom input component for a sell transaction and matches the snapshot 1`] = `
<div>
  <h3>
    Select Amount to  sell
  </h3>
  <label>
    BTC
  </label>
  <TextInput
    base="base"
    className="input-quantity-custom "
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="basevalue"
    onChange={[Function]}
    placeholder="0"
    placeholderParam={Object {}}
    translation="BUY_SELL_CUSTOM"
    type="text"
    value="1"
    writeOnce={false}
  />
  <label>
    USD
  </label>
  <TextInput
    className=""
    disabled={true}
    errorMessages={Array []}
    hideInput={false}
    name="quotevalue"
    onChange={[Function]}
    placeholder="0"
    placeholderParam={Object {}}
    translation="BUY_SELL_CUSTOM"
    type="text"
    value="3000"
    writeOnce={false}
  />
  <p>
    Enter the amount of BTC you wish to  sell. You may enter the amount either directly in BTC units, or in the equivalent amount of USD.
  </p>
</div>
`;
